packages/aws-cdk-lib/aws-ec2/lib/vpc-flow-logs.ts

```diff
@@ -800,6 +829,17 @@ export class FlowLog extends FlowLogBase {
       }).join(' ');
     }
 
+    let trafficType: FlowLogTrafficType | undefined = props.trafficType ?? FlowLogTrafficType.ALL;
+    if (props.resourceType.resourceType === 'TransitGateway' || props.resourceType.resourceType === 'TransitGatewayAttachment') {
```

Comment:
Can you create constants or enums for the `resourceType`?

Reply:
@paulhcsun 
I agree with creating an enum! 
However, in the existing code, a `resourceType` defined as a string is already being used. Along with defining the enum, is it okay to also modify the `FlowLogResourceType`'s `ResourceType` variable and the methods `fromSubnet()`, `fromVpc()`, and `fromNetworkInterfaceId()`?

```ts
const enum ResourceType {
  TRANSIT_GATEWAY = 'transitGateway',
  ...
}

export abstract class FlowLogResourceType {
  public static fromSubnet(subnet: ISubnet): FlowLogResourceType {
    return {
      resourceType: ResourceType.SUBNET,
      resourceId: subnet.subnetId,
    };
  }
...
  /**
   * The type of resource to attach a flow log to.
   */
  public abstract resourceType: ResourceType;
...
}

```

Reply:
@paulhcsun I see! I may be create PR for that later.

Reply:
@badmintoncryer I don't think doing so would be safe so I would hold off on that for now

Reply:
@kaizencc Understood. Thank you for the additional information.